common/src/main/java/com/microsoft/identity/common/internal/providers/ropc/ResourceOwnerPasswordCredentialsTestStrategy.java

```diff
@@ -0,0 +1,146 @@
+package com.microsoft.identity.common.internal.providers.ropc;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.logging.DiagnosticContext;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.net.HttpResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsAuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsAuthorizationResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Configuration;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsTokenRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStrategy;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenResult;
+import com.microsoft.identity.common.internal.result.ResultFuture;
+import com.microsoft.identity.common.internal.util.StringUtil;
+import com.microsoft.identity.common.utilities.Scenario;
+
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.Future;
+
+public class ResourceOwnerPasswordCredentialsTestStrategy extends MicrosoftStsOAuth2Strategy {
```

Comment:
Is it possible to create one generic mock test strategy, that allows us to inject different values for each strategies somewhere i.e. arguments.

Reply:
What kind of values are you talking about injecting? Are you referring to the mock responses?